hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -169,6 +173,12 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
         tbl.getParameters().put("EXTERNAL", "TRUE"); // using the external table type also requires this
       }
 
+      // If needed set the 'storate_handler' property to enable query from Hive
+      if (hiveEngineEnabled) {
+        tbl.getParameters().put(hive_metastoreConstants.META_TABLE_STORAGE,
+            "org.apache.iceberg.mr.hive.HiveIcebergStorageHandler");
```

Comment:
Should this be removed if the engine is not enabled? Because this is a table property, I don't think it would be automatically removed at any point.

Reply:
Good catch!
Done